src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/IdentifiersShouldNotContainUnderscoresTests.cs

```diff
@@ -140,33 +223,36 @@ internal class C2
     }
 }
 ",
-            GetCA1707CSharpResultAt(line: 4, column: 26, symbolKind: SymbolKind.Member, identifierNames: "DoesNotMatter.ConstField_"),
-            GetCA1707CSharpResultAt(line: 5, column: 36, symbolKind: SymbolKind.Member, identifierNames: "DoesNotMatter.StaticReadOnlyField_"),
-            GetCA1707CSharpResultAt(line: 16, column: 5, symbolKind: SymbolKind.Member, identifierNames: "DoesNotMatterEnum._EnumWithUnderscore"),
-            GetCA1707CSharpResultAt(line: 17, column: 5, symbolKind: SymbolKind.Member, identifierNames: "DoesNotMatterEnum._"));
+                    },
+                    ExpectedDiagnostics =
+                    {
+                        VerifyCS.Diagnostic(IdentifiersShouldNotContainUnderscoresAnalyzer.MemberRule).WithLocation(0).WithArguments("DoesNotMatterEnum._"),
+                    },
+                },
+            }.RunAsync();
         }
 
         [Fact, WorkItem(1432, "https://github.com/dotnet/roslyn-analyzers/issues/1432")]
         public async Task CA1707_ForMethods_CSharp()
         {
-            await VerifyCS.VerifyAnalyzerAsync(@"
+            await VerifyCS.VerifyCodeFixAsync(@"
 public class DoesNotMatter
 {
-    public void PublicM1_() { }
+    public void {|#0:PublicM1_|}() { }
     private void PrivateM2_() { } // No diagnostic
```

Comment:
Interesting - I presume this is due to the fact that this rule only runs on public API surface by default? If so, we should have a follow-up PR to make this rule support [api_surface](https://github.com/dotnet/roslyn-analyzers/blob/master/docs/Analyzer%20Configuration.md#analyzed-api-surface) option. Implementation would be as simple as invoking `symbol.MatchesConfiguredVisibility()` extension method instead of `symbol.IsExternallyVisible()`

Reply:
I have just checked the rule and we already use `MatchesConfiguredVisiblity`, I will close the ticket.